corral/cmd/cmd_fetch.pony

```diff
@@ -4,59 +4,83 @@ use "../bundle"
 use "../util"
 use "../vcs"
 
-class CmdFetch
-  let ctx: Context
-
-  new create(ctx': Context, cmd: Command) =>
-    ctx = ctx'
+primitive CmdFetch
+  fun apply(ctx: Context, cmd: Command) =>
     //ctx.log.info("fetch: " + cmd.string())
     ctx.env.out.print("\nfetch:")
-    match BundleFile.load_bundle(ctx.env, ctx.directory, ctx.log)
-    | let bundle: Bundle =>
-      fetch_bundle_deps(bundle, bundle)
+    match BundleFile.load_bundle(ctx.bundle_dir, ctx.log)
+    | let bundle: Bundle val =>
+      _Fetcher(ctx, bundle).fetch_bundle_deps(bundle)
     | let err: Error =>
       ctx.env.out.print(err.message)
       ctx.env.exitcode(1)
     end
 
-  fun fetch_bundle_deps(base_bundle: Bundle box, bundle: Bundle) =>
-    ctx.log.info("Fetching bundle: " + bundle.name())
+actor _Fetcher
+  let ctx: Context
+  let base_bundle: Bundle val
+
+  new create(ctx': Context, base_bundle': Bundle val) =>
+    ctx = ctx'
+    base_bundle = base_bundle'
+
+  be fetch_bundle_deps(bundle: Bundle val) =>
+    ctx.log.info("Fetching direct deps of bundle: " + bundle.name())
     for dep in bundle.deps.values() do
-      try
+      //try
         ctx.log.info("Fetching dep: " + dep.name() + " @ " + dep.version())
           //dep.data.locator)
-        fetch_dep(base_bundle, dep)?
-      else
-        ctx.log.err("Error fetching dep: " + dep.name() + " @ " + dep.version())
-          //dep.data.locator)
-      end
+        fetch_dep(dep)
+      //else
+      //  ctx.log.err("Error fetching dep: " + dep.name() + " @ " + dep.version())
+      //    //dep.data.locator)
+      //end
     end
-    for dep in bundle.deps.values() do
-      // TODO: prevent infinite recursion by keeping & checking a deps Set
-      try
-        let bundle_dir = base_bundle.dep_bundle_root(dep)?
-        ctx.log.fine("Fetching dep's bundle in: " + bundle_dir.path)
-        let dep_bundle = Bundle.load(ctx.env, bundle_dir, ctx.log)?
-        ctx.log.fine("Fetched dep's bundle is: " + dep_bundle.name())
-        fetch_bundle_deps(base_bundle, dep_bundle)
-      else
-        ctx.log.err("Error loading/fetching dep's bundle: " + dep.flat_name())
-      end
+    //fetch_transitive_deps(bundle)
+
+  be fetch_transitive_dep(dep: Dep val) =>
+    ctx.log.info("Fetching transitive dep: " + dep.name())
+    // TODO: prevent infinite recursion by keeping & checking a deps Set
```

Comment:
an issue should be opened for this.

Reply:
https://github.com/ponylang/corral/issues/62